common/util/vector_tree_iterators.h

```diff
@@ -0,0 +1,182 @@
+// Copyright 2017-2021 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef VERIBLE_COMMON_UTIL_VECTOR_TREE_ITERATORS_H_
+#define VERIBLE_COMMON_UTIL_VECTOR_TREE_ITERATORS_H_
+
+#include <iterator>
+
+#include "common/util/iterator_range.h"
+#include "common/util/logging.h"
+#include "common/util/vector_tree.h"
+
```

Comment:
Maybe a little file header comment for someone scanning the file to see that this provides iterators to traverse VectorTrees in various ways ?

Reply:
Done